src/main/java/com/google/googleidentity/oauth2/exception/InvalidGrantException.java

```diff
@@ -0,0 +1,53 @@
+/*
+    Copyright 2020 Google LLC
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package com.google.googleidentity.oauth2.exception;
+
+public class InvalidGrantException extends OAuth2Exception{
+
+    private static final String INVALID_GRANT = "invalid_grant";
+
+    public InvalidGrantException(ErrorCode errorCode) {
```

Comment:
Since this only accept a subset of ErrorCode and ErrorCode in other fixed-errorCode exceptions like InvalidScopeException is not used, maybe we can move the certain subset of ErrorCode to each multi-errorCode classes? So callers won't need to go to the details of these classes to see which error code shall or shall not be set.